staging/src/k8s.io/kubectl/pkg/cmd/create/create_secret_tls.go

```diff
@@ -228,16 +233,41 @@ func (o *CreateSecretTLSOptions) createSecretTLS() (*corev1.Secret, error) {
 	if err != nil {
 		return nil, err
 	}
+	var tlsCertAuthority []byte
+	if len(o.CertAuthority) != 0 {
+		tlsCertAuthority, err = readFile(o.CertAuthority)
+		if err != nil {
+			return nil, err
+		}
+	}
 	if _, err := tls.X509KeyPair(tlsCert, tlsKey); err != nil {
 		return nil, err
 	}
+
+	if len(tlsCertAuthority) > 0 {
+		// validate certAuthority is a valid certificate
+		var block *pem.Block
+		for rest := tlsCertAuthority; len(rest) > 0; {
```

Comment:
Just found this https://github.com/kubernetes/kubernetes/blob/22d120b4d68c8993f2782b9020b230e15f626c3c/staging/src/k8s.io/client-go/util/cert/io.go#L103

Does it help here?

Reply:
The biggest problem from this function is that it does not validate if the pem is actually a certificate. PEM can be a PRIVATE KEY, a CERTIFICATE and some other stuffs. But when you are configuring a CA, you expect that all of the contents in the file are CERTIFICATES.

But, we could make some use in a future of this extraction of the code to other parts (like the other PR you've marked me) so maybe we can move this certificate validation to cmd/utils :)